pkg/prometheus/api.go

```diff
@@ -91,7 +93,10 @@ func (a API) queryWithCache(ctx context.Context, query string, t time.Time) (mod
 	if err != nil {
 		return nil, err
 	}
+
+	a.lock.Lock()
 	a.cache[query] = cachedValue{val, time.Now()}
+	a.lock.Unlock()
```

Comment:
```suggestion
	defer a.lock.Unlock()
	a.cache[query] = cachedValue{val, time.Now()}
```
shouldn't be any panics here, but it's best practice iirc